ggml-cuda.cu

```diff
@@ -51,7 +51,14 @@ static_assert(sizeof(half) == sizeof(ggml_fp16_t), "wrong fp16 size");
             exit(1);                                                                    \
         }                                                                               \
     } while (0)
-#endif // CUDART_VERSION >= 11
+#endif // CUDART_VERSION >= 12000
+
+// define nop for old CUDA versions to fix compilation issues
+#if CUDART_VERSION < 11020
+__device__ void __builtin_assume(bool exp) {
```

Comment:
Why not do something like this?
```suggestion
#define __builtin_assume(exp) do { if (exp) __builtin_unreachable(); } while (0)
```

Reply:
From what I can tell __builtin_unreachable was only introduced with CUDA 11.3 so I don't see how that would help. More generally I don't want to mess with the code for recent CUDA versions if at all possible and would rather push the special cases onto the old CUDA versions that far fewer people use.

Reply:
Oh, I didn't realize CUDA didn't have that feature until recently, since GCC has had it since 4.5.

Reply:
I see.
```
ggml-cuda.cu(1963): error: calling a __host__ function("__builtin_unreachable") from a __device__ function("load_tiles_q4_0<(int)64, (int)8, (bool)0> ") is not allowed
```